### PR TITLE
Them20

### PR DESCRIPTION
**Pull request by nfosdick1**
Adding cpu monitoring when system is spun up.

### CODE REVIEW

File: virtual_machine/windows/main.tf [hunk 1]
```diff
@@ -74,3 +74,81 @@ resource "azurerm_virtual_machine" "virtual_machine" {
     builtby = "Terraform"
   }
 }
+
+resource "azurerm_metric_alertrule" "metric_alertrule_cpu" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-cpu"
```
Comment (ghost) [inline virtual_machine/windows/main.tf]: Should this portion that's hard-coded where it says "_vm_" be the "metric_alertrule" from the resource type?

File: virtual_machine/windows/main.tf [hunk 2]
```diff
@@ -74,3 +74,81 @@ resource "azurerm_virtual_machine" "virtual_machine" {
...
+
+    custom_emails = [
+      "${var.support_email}",
+    ]
+  }
+}
+
+resource "azurerm_metric_alertrule" "metric_alertrule_memory" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-memory"
```
Comment (ghost) [inline virtual_machine/windows/main.tf]: Same as above, is the hardcoded "_vm_" supposed to be the name of the rule to make the resource more accurate?

File: virtual_machine/windows/main.tf [hunk 3]
```diff
@@ -74,3 +74,81 @@ resource "azurerm_virtual_machine" "virtual_machine" {
...
+  resource_group_name = "${var.resource_group}"
+  location            = "${var.location}"
+
+  description = "An alert rule to watch the metric Percentage Memory"
+
+  enabled = true
+
+  resource_id = "${element(azurerm_virtual_machine.virtual_machine.*.id, count.index + var.start_index)}"
+  metric_name = "\\Memory\\Available Bytes"
```
Comment (ghost) [inline virtual_machine/windows/main.tf:115]: What's with the backslashes here?

File: virtual_machine/windows/main.tf [hunk 4]
```diff
@@ -74,3 +74,81 @@ resource "azurerm_virtual_machine" "virtual_machine" {
...
+
+    custom_emails = [
+      "${var.support_email}",
+    ]
+  }
+}
+
+resource "azurerm_metric_alertrule" "metric_alertrule_diskusage" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-diskusage"
```
Comment (ghost) [inline virtual_machine/windows/main.tf]: Same as above regarding the "_vm_" part of the resource name.

File: virtual_machine/windows/main.tf [hunk 5]
```diff
@@ -74,3 +74,81 @@ resource "azurerm_virtual_machine" "virtual_machine" {
...
+  resource_group_name = "${var.resource_group}"
+  location            = "${var.location}"
+
+  description = "An alert rule to watch the metric Percentage Diskusage"
+
+  enabled = true
+
+  resource_id = "${element(azurerm_virtual_machine.virtual_machine.*.id, count.index + var.start_index)}"
+  metric_name = "\\LogicalDisk(_Total)\\% Free Space"
```
Comment (ghost) [inline virtual_machine/windows/main.tf:141]: Are these backslashes necessary escape characters for something?

File: virtual_machine/windows_private/main.tf [hunk 6]
```diff
@@ -61,3 +61,29 @@ resource "azurerm_virtual_machine" "virtual_machine" {
     builtby = "Terraform"
   }
 }
+
+resource "azurerm_metric_alertrule" "metric_alertrule" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-cpu"
```
Comment (ghost) [inline virtual_machine/windows_private/main.tf]: Same as above for the "_vm_" portion.

File: virtual_machine/windows_private/main.tf [hunk 7]
```diff
@@ -61,3 +61,29 @@ resource "azurerm_virtual_machine" "virtual_machine" {
...
+  operator    = "GreaterThan"
+  threshold   = 90
+  aggregation = "Average"
+  period      = "PT5M"
+
+  email_action {
+    send_to_service_owners = false
+
+    custom_emails = [
+      "support@themisds.com",
```
Comment (ghost) [inline virtual_machine/windows_private/main.tf]: This is hard-coded whereas the others are variables, also, you should remove the e-mail address from this public shared repo before I merge this in.
Can you rebase over this to remove it from the git history before it gets re-submitted?

**Review by ghost — CHANGES_REQUESTED** [review]
Please remove client information from the private repo, rebase, and re-submit, once that's done I will approve/merge. Looks good overall.

File: virtual_machine/windows/main.tf (hunk 1 above)
Comment (nfosdick1) [reply virtual_machine/windows/main.tf]: fixing name standard issues.

File: virtual_machine/windows/main.tf (hunk 2 above)
Comment (nfosdick1) [reply virtual_machine/windows/main.tf]: fixing name standard issues.

File: virtual_machine/windows/main.tf (hunk 3 above)
Comment (nfosdick1) [reply virtual_machine/windows/main.tf:115]: The back slashes are required to escape them the backslash.  If you try it with a single terraform will not run successfully.

File: virtual_machine/windows/main.tf (hunk 4 above)
Comment (nfosdick1) [reply virtual_machine/windows/main.tf]: fixing name standard issues.

File: virtual_machine/windows/main.tf (hunk 5 above)
Comment (nfosdick1) [reply virtual_machine/windows/main.tf:141]: Yes they are.

Comment (nfosdick1) [reply virtual_machine/windows/main.tf:141]: fixing name standard issues.

File: virtual_machine/windows_private/main.tf (hunk 6 above)
Comment (nfosdick1) [reply virtual_machine/windows_private/main.tf]: fixing name standard issues.

**Review by ghost — APPROVED** [review]
Approved.

**Pull request merged**